Port upstream test_torch updates into test_torch_xpu.py and update XPU adaptations.

libohao1201 opened the pull request:
disable_e2e
disable_distribute

libohao1201 committed:
1. Modified methods in file torch/testing/_internal/common_utils.py:
- Modified get_all_device_types to include intel GPU if it is available.
- Modified TestCase.wrap_with_cuda_policy to include intel GPU device.
- Modified TestCase._should_stop_test_suite to consider intel GPU device.
- Added flag TEST_XPU_MEM_LEAK_CHECK to TestEnvironment for memory leak checking on intel GPU.
2. Modified Class TestVitalSignsCuda and TestDevicePrecision
- Use decorator onlyOn to test both 'cuda' and intel GPU.

libohao1201 committed:
Modify class TestTorch: add unit tests for torch.xpu operations

libohao1201 committed:
Fixes a decorator onlyOn usage issue

libohao1201 committed:
Modified class TestTorchDeviceType to include allow_xpu=True in instantiate_device_type_tests

libohao1201 committed:
Add xpu in device test list

libohao1201 committed:
Skip cases that xpu not supported or has issues.

libohao1201 committed:
Skip test_index_add/test_index_add_all_dtypes/test_index_add_correctness for XPU

libohao1201 committed:
Port xpu tests.

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Merge conflict

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Remove function test_doc in TestTorch (Lint issue)

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Redefine get_all_device_types for XPU support

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Revise according to the comment.

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Only test xpu for TestDevicePrecision.

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Skip xpu on test_torch_xpu.TestTorchDeviceTypeXPU,test_grad_scaler_deprecated_warning_xpu as xpu not support this old API. See https://github.com/intel/torch-xpu-ops/issues/2509.

libohao1201 committed:
Fix lint error.

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

chuanqi129 commented:
@copilot The Python lint (flake8/ruff) check failed. Please fix the lint errors in this PR.

Lint job log: https://github.com/intel/torch-xpu-ops/actions/runs/25845036291

Instructions:
1. Review the lint failure log via the link above
2. Fix all lint errors in the relevant source files
3. Ensure both Python lint and Clang format checks pass

Copilot committed:
Fix pyfmt spacing in xpu torch test

Agent-Logs-Url: https://github.com/intel/torch-xpu-ops/sessions/065cbb67-a13d-4350-bad8-994883cf1503

Co-authored-by: chuanqi129 <13608516+chuanqi129@users.noreply.github.com>

BBBela commented on test/xpu/test_torch_xpu.py:
```diff
@@ -2122,7 +2074,7 @@ def test_nondeterministic_alert_NLLLoss(self, device):
 
         self.check_nondeterministic_alert(
```
Suggestion:
```python
device_secific_alerts = {
    "cuda": "nll_loss2d_forward_out_cuda_template",
    "xpu": "nll_loss2d_forward_xpu",
}
self.check_nondeterministic_alert(
    lambda: module(input, target),
    device_secific_alerts[torch.device(device).type],
    torch.device(device).type == "cuda" or torch.device(device).type == "xpu",
)
```
Fixes: https://github.com/intel/torch-xpu-ops/issues/2517

BBBela commented on test/xpu/test_torch_xpu.py:
```diff
@@ -2138,7 +2090,7 @@ def test_nondeterministic_alert_CTCLoss(self, device):
 
         self.check_nondeterministic_alert(
```
Suggestion:
```python
device_secific_alerts = {
    "cuda": "ctc_loss_backward_gpu",
    "xpu": "ctc_loss_backward_xpu",
}
self.check_nondeterministic_alert(
    lambda: res.backward(grad, retain_graph=True),
    device_secific_alerts[torch.device(device).type],
    torch.device(device).type == "cuda" or torch.device(device).type == "xpu",
)
```
Fixes: https://github.com/intel/torch-xpu-ops/issues/2514

BBBela commented on test/xpu/test_torch_xpu.py:
```diff
@@ -2414,28 +2390,20 @@ def test_func(call_type):
-                    "median CUDA with indices output",
-                    should_error,
-                )
+            self.check_nondeterministic_alert(
```
Suggestion:
```python
self.check_nondeterministic_alert(
    lambda: test_func(call_type),
    f"median {torch.device(device).type.upper()} with indices output",
    should_error,
)
```
Fixes: https://github.com/intel/torch-xpu-ops/issues/2516

BBBela reviewed (commented):
Hello @libohao1201,
I am writing regarding three issues you recently reopened:
https://github.com/intel/torch-xpu-ops/issues/2517, https://github.com/intel/torch-xpu-ops/issues/2516 and https://github.com/intel/torch-xpu-ops/issues/2514

All of them are caused by compared strings mismatch. All 3 test cases works correctly, just the test function checks the exact strings. Your PR changes those strings, thus causing those test cases to fail while the raised alerts strings hasn't changed.

Below I attach three changes that would fix the three mentioned issues - they just adjust the strings returned by XPU to test function.

libohao1201 committed:
Revise according to review comments.

chuanqi129 commented:
@copilot The Python lint (flake8/ruff) check failed. Please fix the lint errors in this PR.

**Important: Do NOT update the plan in the PR description directly. Use reply comments to update the status.**

Lint job log: https://github.com/intel/torch-xpu-ops/actions/runs/26139332870

Instructions:
1. Review the lint failure log via the link above
2. Fix all lint errors in the relevant source files
3. Ensure both Python lint and Clang format checks pass
4. Use reply comments to report progress - do NOT modify the PR description plan directly

Copilot committed:
Fix trailing-whitespace lint in test_torch_xpu

Agent-Logs-Url: https://github.com/intel/torch-xpu-ops/sessions/98d4b061-adac-4903-ae8c-f20ae673616e

Co-authored-by: chuanqi129 <13608516+chuanqi129@users.noreply.github.com>

chuanqi129 commented:
@copilot New UT failures detected in **op_ut** tests (266 total). Please check these new failures and analyze whether they are caused by the PR changes.

**Important: Do NOT update the plan in the PR description directly. Use reply comments to update the status.**

Job log: https://github.com/intel/torch-xpu-ops/actions/runs/26139521281
Full new failure report: https://github.com/intel/torch-xpu-ops/actions/runs/26139521281/artifacts/7104355214

**New failures:**

| Category | Class name | Test name | Status | Message |
| --- | --- | --- | --- | --- |
| op_ut |  | third_party.torch-xpu-ops.test.xpu.dynamo.test_ctx_manager_xpu | error | AttributeError: module 'torch._dynamo.test_case' has no attribute 'TestCaseWithNestedGraphBreaks'
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutograd | test_mark_activations_dynamic | failed | AssertionError: Expected hasattr(..., '_dynamo_weak_dynamic_indices') to be True
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutograd | test_mark_activations_dynamic_with_nested | failed | AssertionError: Expected hasattr(..., '_dynamo_weak_dynamic_indices') to be True
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutogradWithCache | test_mark_activations_dynamic | failed | AssertionError: Expected hasattr(..., '_dynamo_weak_dynamic_indices') to be True
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutogradWithCache | test_mark_activations_dynamic_with_nested | failed | AssertionError: Expected hasattr(..., '_dynamo_weak_dynamic_indices') to be True

... and 261 more failures. See full list in artifact.

Instructions:
1. Download the full new failure report artifact for complete error messages
2. Analyze whether the failures are caused by PR changes
3. If caused by PR changes, fix those failures on this PR branch directly
4. Add failing test cases with pytest commands in the PR description for the reproducer job to use
5. Use reply comments to report analysis status - do NOT modify the PR description plan directly

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

chuanqi129 commented:
@copilot New UT failures detected in **op_ut** tests (265 total). Please check these new failures and analyze whether they are caused by the PR changes.

**Important: Do NOT update the plan in the PR description directly. Use reply comments to update the status.**

Job log: https://github.com/intel/torch-xpu-ops/actions/runs/26201402012
Full new failure report: https://github.com/intel/torch-xpu-ops/actions/runs/26201402012/artifacts/7128212611

**New failures:**

| Category | Class name | Test name | Status | Message |
| --- | --- | --- | --- | --- |
| op_ut |  | third_party.torch-xpu-ops.test.xpu.dynamo.test_ctx_manager_xpu | error | AttributeError: module 'torch._dynamo.test_case' has no attribute 'TestCaseWithNestedGraphBreaks'
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutogradWithCache | test_mark_outputs_dynamic_use_autograd_True | failed | AssertionError: False is not true
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutograd | test_mark_outputs_dynamic_use_autograd_True | failed | AssertionError: False is not true
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutogradWithDynamo | test_mark_outputs_dynamic_use_autograd_True | failed | AssertionError: False is not true
| op_ut | third_party.torch-xpu-ops.test.xpu.functorch.test_aotdispatch_xpu.TestAOTAutograd | test_mark_activations_dynamic | failed | AssertionError: Expected hasattr(..., '_dynamo_weak_dynamic_indices') to be True

... and 260 more failures. See full list in artifact.

Instructions:
1. Download the full new failure report artifact for complete error messages
2. Analyze whether the failures are caused by PR changes
3. If caused by PR changes, fix those failures on this PR branch directly
4. Add failing test cases with pytest commands in the PR description for the reproducer job to use
5. Use reply comments to report analysis status - do NOT modify the PR description plan directly

libohao1201 committed:
Merge branch 'main' into libo/ut_test_torch

chunhuanMeng commented on test/xpu/test_torch_xpu.py:
```diff
@@ -2216,8 +2228,12 @@ def test_nondeterministic_alert_histc(self, device, dtype):
                 lambda: op_call(a, min=0, max=3),
-                "_histc_xpu with floating point input",
-                torch.device(device).type == "xpu" and dtype.is_floating_point,
+                "_histc_" + torch.device(device).type + "_with floating point input",
```
please change the `_with floating point input` to ` with floating point input` — replacing the underscore `_` before "with" with a space.

libohao1201 committed:
Revise according to the comment.

libohao1201 committed:
Revise scripts of test_nondeterministic_alert_histc.
